### DISCUSSION TITLE
Moving from docker to shared hosting environment

### DISCUSSION THREAD

**Discussion by mvandenabeele** [shlinkio/shlink#1956] [Help wanted] [answered]
### Shlink version

3.7.0

### PHP version

8.3

### How do you serve Shlink

Self-hosted Apache

### Database engine

MySQL

### Database version

5.7.40

### Summary

I have shlink instance running for quite some time now but I wanted to move it to a shared hosting platform for better availability. I was able to copy the database to the new server, and run vendor/bin/shlink-installer on the new server to configure the database. The shortlinks work as expected, but shlink-ui can't talk to this new server. I always get a 500 internal server error, but nothing appears in the server logs. How can I know what is going on and how can I fix this?

**Comment by acelaya** [comment]
500 errors with no logs usually mean that the user used by Shlink in the web context, does not have permissions to write in the `data` folder.

Grant write permissions recursively and it should start working.

**Comment by mvandenabeele** [comment]
Thanks for the speedy reply. I was looking for logs in the folder where alle PHP errors are dumped. Now, I found the location where shlink is putting the logs, and those files are written. I guess permissions are okay then. 

This is the last entry in the log:

`[2023-12-19T12:31:08.148737+01:00] [ba43ce52-bcb0-4809-bb7f-ee1900cf93c2] Shlink.ERROR - ArgumentCountError: Too few arguments to function Shlinkio\Shlink\Rest\Middleware\AuthenticationMiddleware::__construct(), 0 passed in /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/mezzio/mezzio/src/MiddlewareContainer.php on line 53 and exactly 3 expected in /data/sites/web/jabblacloud/subsites/url.jabbla.com/module/Rest/src/Middleware/AuthenticationMiddleware.php:26
Stack trace:
#0 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/mezzio/mezzio/src/MiddlewareContainer.php(53): Shlinkio\Shlink\Rest\Middleware\AuthenticationMiddleware->__construct()
#1 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/mezzio/mezzio/src/Middleware/LazyLoadingMiddleware.php(26): Mezzio\MiddlewareContainer->get('...')
#2 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/Next.php(49): Mezzio\Middleware\LazyLoadingMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#3 /data/sites/web/jabblacloud/subsites/url.jabbla.com/module/Rest/src/Middleware/BodyParserMiddleware.php(34): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
#4 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/mezzio/mezzio/src/Middleware/LazyLoadingMiddleware.php(26): Shlinkio\Shlink\Rest\Middleware\BodyParserMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#5 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/Next.php(49): Mezzio\Middleware\LazyLoadingMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#6 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/mezzio/mezzio-router/src/Middleware/ImplicitOptionsMiddleware.php(75): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
#7 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/mezzio/mezzio/src/Middleware/LazyLoadingMiddleware.php(26): Mezzio\Router\Middleware\ImplicitOptionsMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#8 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/Next.php(49): Mezzio\Middleware\LazyLoadingMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#9 /data/sites/web/jabblacloud/subsites/url.jabbla.com/module/Rest/src/Middleware/ErrorHandler/BackwardsCompatibleProblemDetailsHandler.php(22): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
#10 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/mezzio/mezzio/src/Middleware/LazyLoadingMiddleware.php(26): Shlinkio\Shlink\Rest\Middleware\ErrorHandler\BackwardsCompatibleProblemDetailsHandler->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#11 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/Next.php(49): Mezzio\Middleware\LazyLoadingMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#12 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/MiddlewarePipe.php(75): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
#13 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/Middleware/PathMiddlewareDecorator.php(60): Laminas\Stratigility\MiddlewarePipe->process(Object(Laminas\Diactoros\ServerRequest), Object(Psr\Http\Server\RequestHandlerInterface@anonymous))
#14 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/Next.php(49): Laminas\Stratigility\Middleware\PathMiddlewareDecorator->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#15 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/Next.php(42): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
#16 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/mezzio/mezzio-router/src/Middleware/ImplicitHeadMiddleware.php(77): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
#17 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/mezzio/mezzio/src/Middleware/LazyLoadingMiddleware.php(26): Mezzio\Router\Middleware\ImplicitHeadMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#18 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/Next.php(49): Mezzio\Middleware\LazyLoadingMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#19 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/mezzio/mezzio-router/src/Middleware/RouteMiddleware.php(50): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
#20 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/mezzio/mezzio/src/Middleware/LazyLoadingMiddleware.php(26): Mezzio\Router\Middleware\RouteMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#21 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/Next.php(49): Mezzio\Middleware\LazyLoadingMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#22 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/MiddlewarePipe.php(75): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
#23 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/Next.php(49): Laminas\Stratigility\MiddlewarePipe->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#24 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/Next.php(42): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
#25 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/shlinkio/shlink-common/src/Middleware/CloseDbConnectionMiddleware.php(24): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
#26 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/mezzio/mezzio/src/Middleware/LazyLoadingMiddleware.php(26): Shlinkio\Shlink\Common\Middleware\CloseDbConnectionMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#27 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/Next.php(49): Mezzio\Middleware\LazyLoadingMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#28 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/MiddlewarePipe.php(75): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
#29 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/Next.php(49): Laminas\Stratigility\MiddlewarePipe->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#30 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/Middleware/PathMiddlewareDecorator.php(122): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
#31 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/Next.php(42): Psr\Http\Server\RequestHandlerInterface@anonymous->handle(Object(Laminas\Diactoros\ServerRequest))
#32 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/mezzio/mezzio-problem-details/src/ProblemDetailsMiddleware.php(46): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
#33 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/mezzio/mezzio/src/Middleware/LazyLoadingMiddleware.php(26): Mezzio\ProblemDetails\ProblemDetailsMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#34 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/Next.php(49): Mezzio\Middleware\LazyLoadingMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#35 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/MiddlewarePipe.php(75): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
#36 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/Middleware/PathMiddlewareDecorator.php(60): Laminas\Stratigility\MiddlewarePipe->process(Object(Laminas\Diactoros\ServerRequest), Object(Psr\Http\Server\RequestHandlerInterface@anonymous))
#37 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/Next.php(49): Laminas\Stratigility\Middleware\PathMiddlewareDecorator->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#38 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/Next.php(42): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
#39 /data/sites/web/jabblacloud/subsites/url.jabbla.com/module/Rest/src/Middleware/CrossDomainMiddleware.php(24): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
#40 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/mezzio/mezzio/src/Middleware/LazyLoadingMiddleware.php(26): Shlinkio\Shlink\Rest\Middleware\CrossDomainMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#41 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/Next.php(49): Mezzio\Middleware\LazyLoadingMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#42 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/Middleware/ErrorHandler.php(129): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
#43 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/mezzio/mezzio/src/Middleware/LazyLoadingMiddleware.php(26): Laminas\Stratigility\Middleware\ErrorHandler->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#44 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/Next.php(49): Mezzio\Middleware\LazyLoadingMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#45 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/php-middleware/request-id/src/RequestIdMiddleware.php(32): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
#46 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/mezzio/mezzio/src/Middleware/LazyLoadingMiddleware.php(26): PhpMiddleware\RequestId\RequestIdMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#47 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/Next.php(49): Mezzio\Middleware\LazyLoadingMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#48 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/shlinkio/shlink-common/src/Middleware/ContentLengthMiddleware.php(24): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
#49 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/mezzio/mezzio/src/Middleware/LazyLoadingMiddleware.php(26): Shlinkio\Shlink\Common\Middleware\ContentLengthMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#50 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/Next.php(49): Mezzio\Middleware\LazyLoadingMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#51 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/shlinkio/shlink-common/src/Middleware/AccessLogMiddleware.php(27): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
#52 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/mezzio/mezzio/src/Middleware/LazyLoadingMiddleware.php(26): Shlinkio\Shlink\Common\Middleware\AccessLogMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#53 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/Next.php(49): Mezzio\Middleware\LazyLoadingMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#54 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/MiddlewarePipe.php(75): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
#55 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/Next.php(49): Laminas\Stratigility\MiddlewarePipe->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
#56 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/MiddlewarePipe.php(75): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
#57 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-stratigility/src/MiddlewarePipe.php(64): Laminas\Stratigility\MiddlewarePipe->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\EmptyPipelineHandler))
#58 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/laminas/laminas-httphandlerrunner/src/RequestHandlerRunner.php(73): Laminas\Stratigility\MiddlewarePipe->handle(Object(Laminas\Diactoros\ServerRequest))
#59 /data/sites/web/jabblacloud/subsites/url.jabbla.com/vendor/mezzio/mezzio/src/Application.php(53): Laminas\HttpHandlerRunner\RequestHandlerRunner->run()
#60 /data/sites/web/jabblacloud/subsites/url.jabbla.com/config/run.php(13): Mezzio\Application->run()
#61 /data/sites/web/jabblacloud/subsites/url.jabbla.com/public/index.php(5): {closure}()
#62 {main}
`

Maybe this stacktrace helps?

**Reply by acelaya** [reply]
This is very strange. This should not happen unless the project configuration gets messed up somehow.

Can you describe exactly the steps you followed to set-up your new Shlink instance?

Also, can you try if this happens with v3.7.1 too?

**Reply by mvandenabeele** [reply, marked as answer]
I must admit I have been trying a lot to move this site from a docker environment to a shared hosting platform. I did read the manual on that subject so I knew I could import all urls from another instance, but I missed that I had SSH access to the server of the shared hosting platform. So I tried going trough the install wizard in a new docker container to get a configuration file, and then I packed all files in the docker container and pushed those using FTP. This morning, I started from scratch following the instructions in the manual and using SSH, and it all went fine. I'm sorry about the noise here!

Thanks for working on this nice product and for taking the time to follow up on support. You're doing a great job!

**Reply by acelaya** [reply]
> I tried going trough the install wizard in a new docker container to get a configuration file, and then I packed all files in the docker container and pushed those using FTP

Yeah, this was the recipe for disaster 😅

I'm glad it all went well though.
